packages/spanner/src/transaction.js

```diff
@@ -283,11 +314,28 @@ Transaction.prototype.commit = function(callback) {
  * @param {function} callback - The callback function.
  */
 Transaction.prototype.request = function(config, callback) {
+  var self = this;
+
   var reqOpts = extend({
     session: this.session.formattedName_
   }, config.reqOpts);
 
-  config.method(reqOpts, callback);
+  var gaxOptions = reqOpts.gaxOptions;
+  delete reqOpts.gaxOptions;
+
+  config.method(reqOpts, gaxOptions, function(err, resp) {
```

Comment:
I don't think letting the same stream mix both invalid and valid response items is an acceptable solution. I think we should be as loud about the failure as possible, and destroy the stream, so the user can decide to retry on their own.

@vkedia we could still use your guidance on when a retry should be performed, as well as how you see that working with a streaming request.